lib/srv/heartbeat.go

```diff
@@ -440,15 +443,39 @@ func (h *Heartbeat) announce() error {
 			if !ok {
 				return trace.BadParameter("expected services.Server, got %#v", h.current)
 			}
-			err := h.Announcer.UpsertKubeService(h.cancelCtx, kube)
+			keepAlive, err := h.Announcer.UpsertKubeServiceV2(h.cancelCtx, kube)
 			if err != nil {
-				h.nextAnnounce = h.Clock.Now().UTC().Add(h.KeepAlivePeriod)
-				h.setState(HeartbeatStateAnnounceWait)
+				// Check if the error is an Unimplemented grpc status code,
+				// if it is fall back to old keepalive method
+				// DELETE in 11.0
+				if e, ok := status.FromError(trail.ToGRPC(err)); ok {
```

Comment:
A switch with only one case seems odd when you already have an if statement. I wonder if something like this would be cleaner:

```
if e, ok := status.FromError(trail.ToGRPC(err)); ok && e.Code() == codes.Unimplemented {
  // ...
}
```